__tests__/components/SignerBaseFilter.spec.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * (C) Symbol Contributors 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+import SignerBaseFilter from '@/components/SignerBaseFilter/SignerBaseFilter.vue';
+import { SignerBaseFilterTs } from '@/components/SignerBaseFilter/SignerBaseFilterTs';
+import { getComponent } from '@MOCKS/Components';
+import { simpleWallet1, simpleWallet2 } from '@MOCKS/Accounts';
+
+describe('components/SignerBaseFilter', () => {
+    const mockSigner = {
+        label: 'wallet 1',
+        address: simpleWallet1.address,
+        multisig: false,
+        requiredCosigApproval: 0,
+        requiredCosigRemoval: 0,
+        parentSigners: [],
+    };
+
+    const mockAccountStore = {
+        namespaced: true,
+        state: {
+            currentSigner: null,
+        },
+        getters: {
+            currentSigner: (state) => {
+                return state.currentSigner;
+            },
+        },
+    };
+
+    const getSignerBaseFilterWrapper = (props = {}) =>
+        getComponent(
+            SignerBaseFilter,
+            {
+                account: mockAccountStore,
+            },
+            {
+                currentSigner: {
+                    ...mockSigner,
+                },
+            },
+            {
+                ...props,
+            },
+            {
+                OptionGroup: true,
+                Icon: true,
+            },
+        );
+
+    describe('component', () => {
+        const runBasicSelectedSignerInMountTests = (props, expectResult) => {
+            // Arrange:
+            const wrapper = getSignerBaseFilterWrapper({
+                rootSigner: mockSigner,
+                ...props,
+            });
+
+            const vm = wrapper.vm as SignerBaseFilterTs;
+
+            // Act + Assert:
+            expect(vm.selectedSigner).toBe(expectResult);
+        };
+
+        test('renders only account in the list when parent signer does not exist', () => {
+            // Arrange:
+            const wrapper = getSignerBaseFilterWrapper({
+                rootSigner: {
+                    label: 'wallet 1',
+                    address: simpleWallet1.address,
+                    multisig: false,
+                    requiredCosigApproval: 0,
+                    requiredCosigRemoval: 0,
+                },
```

Comment:
isn't this mockSigner?

Reply:
yes, but the main difference is that `parantSigners` do not exist.
but I find a better way to write it.

`const { parentSigners, ...rest } = mockSigner;`